x-pack/plugins/apm/public/components/app/service_overview/index.tsx

```diff
@@ -76,6 +84,10 @@ export function ServiceOverview() {
     ? 'column'
     : 'row';
 
+  if (!transactionType && transactionTypeStatus === FETCH_STATUS.SUCCESS) {
+    return <NoTransactionsPrompt />;
+  }
```

Comment:
Maybe I'm missing something here, but we have a couple of chart in the service overview that do not use transaction.type no?

Reply:
Good catch, I checked and we have:
- ServiceOverviewInstancesTable
- InstancesLatencyDistributionChart
- ServiceOverviewErrorsTable

Those components are not `transactionType` dependant.
What do we do? Should we go back to the other approach where we show no data?

https://user-images.githubusercontent.com/1313018/217273382-73aa1987-35cd-4936-b9e0-3377c8fbb066.mov

Maybe if we combine it with a callout it would be more helpful for the user @sqren ?

Reply:
Hiding will cause page jumping no?

Reply:
I think it will cause jumping around. I'd rather have the empty charts and some indication why mostly all of them are empty. wdyt?

Reply:
I'm inclined to agree, but quite possibly @sqren has a better idea

Reply:
Yes, it will cause layout shifts in situations where users don't have transactions. But it won't be a problem in the vast majority of situations imo.

That being said, I'm open to the idea of leaving the charts there. Perhaps we should include @boriskirov since this is a UX question.

Reply:
Unless I'm missing something I don't think this needs to be a pitch. The use case is that sometimes there are no transactions for the selected time range. In those cases we shouldn't show an infinite spinner (because we know they'll never complete). 

@yngrdyn Can you find a simple way to not show the spinner? I felt hacky setting the status like:

```ts
status={transactionType ? mainStatsStatus : FETCH_STATUS.SUCCESS}
```

If we keep the status like this:
```ts
status={mainStatsStatus}
```

Won't it be in the `NOT_INITIATED` state? In which case we can display the empty state (and not the spinner) in that case?

Reply:
Another options is something like
```ts
const isAborted = !!transactionType && transactionTypeStatus === FETCH_STATUS.SUCCESS;

// ...

status={isAborted ? FETCH_STATUS.ABORTED : mainStatsStatus}
```

Reply:
What do you think about the one that I pushed?
As [Cauê suggested](https://github.com/elastic/kibana/pull/150846#issuecomment-1425839154), sort of, I'm checking whether the `transactionType` is undefined and it that case resolving the promise with undefined value. I tried to just reject the promise but in that case the status never changed from `LOADING`.